gr-analog/lib/squelch_base_ff_impl.cc

```diff
@@ -33,13 +33,31 @@ squelch_base_ff_impl::squelch_base_ff_impl(const char* name, int ramp, bool gate
     d_state = ST_MUTED;
     d_envelope = d_ramp ? 0.0 : 1.0;
     d_ramped = 0;
+    precalculate_window();
 }
 
 squelch_base_ff_impl::~squelch_base_ff_impl() {}
 
+void squelch_base_ff_impl::precalculate_window()
+{
+    if (d_ramp <= 0) {
+        d_window_table.clear();
+        return;
+    }
+
+    d_window_table.resize(d_ramp + 1);
+    for (int i = 0; i <= d_ramp; i++) {
+        d_window_table[i] = 0.5 - std::cos(GR_M_PI * i / d_ramp) / 2.0;
```

Comment:
this is a narrowing conversion from `double` to `float`; and we don't need the double precision here, so while we pre-calculate a table, let's also improve this code, use `0.5f` in place of `0.5`, calculate a `const float increment = static_cast<float>(GR_M_PI) / d_ramp`, and use that, as in `std::cos(i * increment)`.